meta-oe/recipes-core/sdbus-c++/sdbus-c++_0.7.1.bb

```diff
@@ -0,0 +1,45 @@
+SUMMARY = "sdbus-c++"
+DESCRIPTION = "High-level C++ D-Bus library designed to provide easy-to-use yet powerful API in modern C++"
+
+SECTION = "libs"
+
+LICENSE = "LGPLv2.1"
+LIC_FILES_CHKSUM = "file://${WORKDIR}/sdbus-cpp-${PV}/COPYING;md5=1803fa9c2c3ce8cb06b4861d75310742"
+
+inherit cmake pkgconfig systemd ptest python3native
+
+DEPENDS = "expat ${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'systemd', 'meson-native ninja-native git-native gperf-native gettext-native util-linux glibc libcap', d)}"
+RDEPENDS_${PN} = "${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'libsystemd', 'glibc libcap', d)}"
+
+SRC_URI = "git://github.com/Kistler-Group/sdbus-cpp.git;protocol=https;destsuffix=sdbus-cpp-${PV};branch=master;tag=v${PV}"
+SRC_URI += "file://run-ptest"
+
+TEST_BIN_DIR = "/opt/test/bin"
+
+# Allow CMake to use git on the host machine
+OECMAKE_FIND_ROOT_PATH_MODE_PROGRAM = "BOTH"
+# Ninja is the default CMake generator in Yocto now, so let's set Unix Makefiles explicitly
+OECMAKE_GENERATOR = "Unix Makefiles"
+
+EXTRA_OECMAKE = "-DBUILD_CODE_GEN=ON \
+                 -DBUILD_DOC=ON \
+                 -DBUILD_DOXYGEN_DOC=OFF \
+                 ${@bb.utils.contains('DISTRO_FEATURES', 'systemd', '-DBUILD_LIBSYSTEMD=OFF', '-DBUILD_LIBSYSTEMD=ON -DLIBSYSTEMD_VERSION=242', d)} \
```

Comment:
hardcoding systemd version is going to be a maintenance pain since it has to be updated whenever systemd in OE-core is updated. Is there any other way to get version ? maybe something like
```pkg-config --modversion systemd```

Reply:
This version has nothing to do with systemd version built by separate recipe(s) in OE-core. It refers to version of just a small part of external libsystemd project that is built into sdbus-c++ library. But I removed that configuration option anyway, since sdbus-c++ has this libsystemd version as a default version in its makefiles.